akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowConflateSpec.scala

```diff
@@ -190,14 +194,19 @@ class FlowConflateSpec extends AkkaSpec {
         .run()
 
       val sub = sourceProbe.expectSubscription()
+
       sub.expectRequest(4)
       sub.sendNext("one")
       sub.sendNext("two")
       sub.sendNext("three")
       sub.sendComplete()
 
+      // delay requesting to avoid race condition
+      sinkProbe.ensureSubscription()
+      sinkProbe.expectNoMsg(100.millis)
```

Comment:
This can be done deterministically and faster by using a TestLatch that is opened within the aggregate function before throwing the exception.